src/compiler/scala/tools/nsc/backend/jvm/analysis/AliasingFrame.scala

```diff
@@ -19,7 +19,7 @@ import AliasSet.SmallBitSet
  *
  * See the doc of package object `analysis` for some notes on the performance of alias analysis.
  */
-class AliasingFrame[V <: Value](nLocals: Int, nStack: Int) extends Frame[V](nLocals, nStack) {
+class AliasingFrame[V <: Value](nLocals: Int, nStack: Int) extends VanillaAsmFrame[V](nLocals, nStack) {
```

Comment:
AliasingFrame is used in copy propagation and nullness analysis, but not in ProdConsAnalyzer. The following diff does that: https://github.com/retronym/scala/compare/bump/asm-6.2...lrytz:2.13.x?expand=1

With that, some tests `ProdConsAnalyzerTest` actually start failing.

There might be a way to implement prod-cons without those additional calls to `copyOperation`, I haven't looked in detail.

Reply:
I would expect `ProdConsAnalyzer` to fail with the `VanillaAsmFrame` until ASM 6.3 which will start to call `newParameterValue`/`newEmptyNonParameterLocalValue`/`newExceptionValue`, which `InitialProducerSourceInterpreter` implements.

We only override `copyOperation` in `NullnessInterpreter`.

Reply:
Not sure I follow.. The `newXValue` methods are called in the Analyzer, not in `Frame` / `VanillaAsmFrame`. The `copyOperation` implementation for ProdCons is inherited from `SourceInterpreter`. Anyway, we'll see when we update to 6.3.

Reply:
Right, I didn't consider the inherited implementation of `copyOperation`, i just looked for the one override in our codebase. Still, it should be easy enough to adapt when we need to.